ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarCustodyImpl.java

```diff
@@ -150,25 +151,40 @@ public SafeFuture<Boolean> hasCustodyDataColumnSidecar(
   @Override
   public void onSlot(final UInt64 slot) {
     currentSlot = slot;
-    if (updateEpoch(spec.computeEpochAtSlot(slot))) {
-      int groupCount = custodyGroupCountManager.getCustodyGroupCount();
-      final int oldGroupCount = totalCustodyGroupCount.getAndSet(groupCount);
-      // TODO-fulu: ignoring the case when it's less, let's skip pruning in early version, to
-      // implement in future invalidating current custody as number of required groups have
-      // increased (https://github.com/Consensys/teku/issues/9468)
-      if (groupCount > oldGroupCount) {
-        LOG.debug("Custody group count changed from {} to {}", oldGroupCount, groupCount);
-        final UInt64 minCustodyPeriodSlot =
-            minCustodyPeriodSlotCalculator.getMinCustodyPeriodSlot(currentSlot);
-        db.setFirstCustodyIncompleteSlot(minCustodyPeriodSlot).ifExceptionGetsHereRaiseABug();
-      }
+    if (!updateEpoch(spec.computeEpochAtSlot(slot))) {
+      return;
+    }
+    final int newCustodyGroupCount = custodyGroupCountManager.getCustodyGroupCount();
+    final int oldCustodyGroupCount = totalCustodyGroupCount.getAndSet(newCustodyGroupCount);
+    if (newCustodyGroupCount == oldCustodyGroupCount) {
+      return;
+    }
+    LOG.debug(
+        "Custody group count changed from {} to {}", oldCustodyGroupCount, newCustodyGroupCount);
+    if (newCustodyGroupCount > oldCustodyGroupCount) {
+      final UInt64 minCustodyPeriodSlot =
+          minCustodyPeriodSlotCalculator.getMinCustodyPeriodSlot(currentSlot);
+      db.setFirstCustodyIncompleteSlot(minCustodyPeriodSlot)
+          .finish(
+              error ->
+                  LOG.error(
+                      "Unexpected error while updating first custody incomplete slot.", error));
     }
   }
 
   @Override
   public void onNewFinalizedCheckpoint(
       final Checkpoint checkpoint, final boolean fromOptimisticBlock) {
-    advanceFirstIncompleteSlot(checkpoint.getEpoch()).ifExceptionGetsHereRaiseABug();
+    advanceFirstIncompleteSlot(checkpoint.getEpoch())
+        .finish(
+            error ->
+                LOG.error(
+                    "Unexpected error while advancing first custody incomplete slot.", error));
```

Comment:
I'd log also checkpoint info and fromOptimisticBlock otherwise it doesn't benefit from log line